### PR TITLE
fix: improve visibility of join button for temporary guests (WPB-3550)

### PR DESCRIPTION
**Pull request by V-Gira**
<table>
<td>
  <a href="https://wearezeta.atlassian.net/browse/WPB-3550" title="WPB-3550" target="_blank">TaskWPB-3550</a>  [WEB] Improve visibility of "join call" button in a guest conversation
  </td></table>
  <br />
 

## Description

When a temporary guest joins a conversation where a call is already ongoing, the join button is easily missed

We decided to show the call ui in that case, even if the call is declined (a temporary guest only has access to 1 conversation, no other groups or 1on1)

## Screenshots/Screencast (for UI changes)

Before:
image

After:
Screenshot from 2024-02-14 17-00-09

## Checklist

- [x] PR has been self reviewed by the author;
- [x] Hard-to-understand areas of the code have been commented;
- [x] If it is a core feature, unit tests have been added;

### CODE REVIEW

File: src/script/components/calling/CallingCell.tsx [hunk 1]
```diff
@@ -304,19 +302,7 @@ const CallingCell: React.FC<CallingCellProps> = ({
...
-          className="call-ui__button call-ui__button--green call-ui__button--join"
-          style={{margin: '40px 16px 0px'}}
-          onClick={() => callActions.answer(call)}
-          type="button"
-          data-uie-name="do-call-controls-call-join"
-        >
-          {t('callJoin')}
-        </button>
-      )}
-
```
Comment (V-Gira) [inline src/script/components/calling/CallingCell.tsx:318]: this is the old join button component, strictly for temporary guests

File: src/script/components/calling/CallingCell.tsx [hunk 2]
```diff
@@ -304,19 +302,7 @@ const CallingCell: React.FC<CallingCellProps> = ({
...
-      {conversation && !isDeclined && (
+      {conversation && (!isDeclined || isTemporaryUser) && (
```
Comment (V-Gira) [inline src/script/components/calling/CallingCell.tsx:305]: we always show the call ui to temp users, even if the call is declined

File: src/script/components/calling/CallingCell.tsx [hunk 3]
```diff
@@ -448,163 +434,170 @@ const CallingCell: React.FC<CallingCellProps> = ({
             <ConversationClassifiedBar conversation={conversation} classifiedDomains={classifiedDomains} />
           )}
 
-          {!isDeclined && (
-            <>
```
Comment (V-Gira) [inline src/script/components/calling/CallingCell.tsx:452]: doesn't appear this check was ever necessary as it's already performed on (previously) line 319

File: src/script/components/calling/CallingCell.tsx [hunk 4]
```diff
@@ -448,163 +434,170 @@ const CallingCell: React.FC<CallingCellProps> = ({
...
+                  {isDeclined ? (
+                    <button
+                      className="call-ui__button call-ui__button--green call-ui__button--join call-ui__button--join--large "
+                      onClick={answerCall}
+                      type="button"
+                      data-uie-name="do-call-controls-call-join"
+                    >
+                      {t('callJoin')}
+                    </button>
+                  ) : (
```
Comment (V-Gira) [inline src/script/components/calling/CallingCell.tsx:557]: `IsDeclined` can only be true at that stage if `ìsTemporaryUser` is also true.

In that case we show a larger join button and no decline button if a call is ongoing bit the temp user hasn't accepted it yet

Comment (atomrc) [reply src/script/components/calling/CallingCell.tsx:557]: damn this logic is quite hard to follow 😅

**Review by atomrc — APPROVED** [review]

File: src/script/components/calling/CallingCell.tsx (hunk 4 above)
Comment (V-Gira) [reply src/script/components/calling/CallingCell.tsx:557]: I could add a few comments, maybe rename
```
const isDeclined = !!reason && [CALL_REASON.STILL_ONGOING, CALL_REASON.ANSWERED_ELSEWHERE].includes(reason);
```
to `isNotAccepted`?

Comment (atomrc) [reply src/script/components/calling/CallingCell.tsx:557]: Nah, I think the problem lies mainly in the rendering part, things are a little too nested. But not sure there is an easy way to fix that now; :)
